.github/workflows/test_ipu.yml

```diff
@@ -0,0 +1,80 @@
+name: test-ipu
+
+on:
+  push:
+    branches: ["main"]
+    tags: ["*"]
+  pull_request:
+    branches:
+      - "*"
+      - "!gh-pages"
+  schedule:
+    - cron: "0 4 * * MON"
+
+jobs:
+  test-ipu:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8"]
+        pytorch-version: ["2.0"]
+
+    runs-on: "ubuntu-latest"
+    timeout-minutes: 300
```

Comment:
Possible to reduce the timeout? (a large upper bound is ok but I guess 300 mins is a bit too much)